### PR TITLE
auto-document cylc modules

### PR DESCRIPTION
**Pull request by oliver-sanders**
Sibling PR: https://github.com/cylc/cylc-doc/pull/148

Auto-generate documentation within Cylc Flow for:

* Jinja2 Filters.
* Batch Sys Handlers.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests (why?).
- [x] No change log entry required (why? e.g. invisible to users).
- [x] Docs updated here - https://github.com/cylc/cylc-doc/pull/148
- [x] No dependency changes.

### CODE REVIEW

**Comment by MetRonnie** [comment]
Tested with the PR branches checked out on all 3 repos

**Comment by oliver-sanders** [comment]
Deconflicted.

**Comment by oliver-sanders** [comment]
That should do it.

File: cylc/flow/cfgspec/suite.py [hunk 1]
```diff
@@ -774,8 +774,8 @@
 
                 The top level share and work directory location can be changed
                 (e.g. to a large data area) by a global config setting (see
-                :cylc:conf:`global.cylc[hosts][<hostname glob>]
-                work directory`).
+                :cylc:conf:`
+                flow.rc[platforms][<platform name>]work directory`).
```
Comment (MetRonnie) [inline cylc/flow/cfgspec/suite.py:778]:
```suggestion
                global.cylc[platforms][<platform name>]work directory`).
```

Comment (MetRonnie) [reply cylc/flow/cfgspec/suite.py:778]: Also the last commit description should be `suite.rc -> flow.cylc`. Otherwise, can't find anything wrong 👍

**Review by MetRonnie — APPROVED** [review]

**Pull request merged**